extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -231,24 +246,30 @@ class ConnectorsManager {
     async _spawnOrDestroyConnector(connector) {
         try {
             if (connector.isRunning && connector.bucketCount === 0) {
+                this.emit('connector-updated', connector);
                 await connector.destroy();
                 this._metricsHandler.onConnectorDestroyed();
                 this._logger.info('Successfully destroyed a connector', {
                     method: 'ConnectorsManager._spawnOrDestroyConnector',
-                    connector: connector.name
+                    connector: connector.name,
                 });
                 return true;
             } else if (!connector.isRunning && connector.bucketCount > 0) {
                 await connector.spawn();
                 this._metricsHandler.onConnectorsInstantiated(false);
                 this._logger.info('Successfully spawned a connector', {
                     method: 'ConnectorsManager._spawnOrDestroyConnector',
-                    connector: connector.name
+                    connector: connector.name,
                 });
                 return true;
-            } else if (connector.isRunning) {
-                return connector.updatePipeline(true);
+            } else if (connector.isRunning && this._allocationStrategy.canUpdate()) {
+                const isPipelineUpdated = connector.updatePipeline(true);
```

Comment:
the function is async, so need to await:
```suggestion
                const isPipelineUpdated = await connector.updatePipeline(true);
```

...however this makes the event be generated after the update was performed, which is not what we want: we may need to send the event from the connector

Reply:
I added the event in the connector, and I eventually moved both in it, so we can handle all events (destroy & update pipeline) from the connector class, and the event is then propagated to the oplog populator through the connectors manager.
I also used a constant to avoid duplication of the string.